crates/polars-plan/src/plans/conversion/dsl_to_ir.rs

```diff
@@ -374,6 +374,58 @@ pub fn to_alp_impl(lp: DslPlan, ctxt: &mut DslConversionContext) -> PolarsResult
             output_schema: None,
             filter: None,
         },
+        // `select(pl.all().filter())` -> `filter()`
+        // `select((pl.all().<exprs>).filter())` -> `select(pl.all().<exprs>).filter()`
+        DslPlan::Select {
+            expr,
+            input,
+            options,
+        } if {
+            // This is a hack so that we don't break `list.eval(pl.element().filter())`. That expression
```

Comment:
Auch, we really need to replace that `element => col("")` hack. With a proper `Expr::Element` type and something that can run it.

But probably not this PR.